src/subdomains/core/history/dto/history.dto.ts

```diff
@@ -34,9 +41,29 @@ export class HistoryDto {
 
   @ApiProperty()
   isComplete: boolean;
+
+  @ApiProperty()
+  status: PaymentStatus;
 }
 
 export class TypedHistoryDto extends HistoryDto {
+  public static PaymentStatusMapper: {
+    [key in BuyCryptoStatus | 'Incomplete']: PaymentStatus;
+  } = {
+    [BuyCryptoStatus.BATCHED]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.CREATED]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.MISSING_LIQUIDITY]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.PAYING_OUT]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.PENDING_LIQUIDITY]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.PREPARED]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.PRICE_MISMATCH]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.PRICE_SLIPPAGE]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.READY_FOR_PAYOUT]: PaymentStatus.PENDING,
+    [BuyCryptoStatus.COMPLETE]: PaymentStatus.COMPLETE,
+    [BuyCryptoStatus.WAITING_FOR_LOWER_FEE]: PaymentStatus.FEE_TOO_HIGH,
+    ['Incomplete']: PaymentStatus.PENDING,
+  };
```

Comment:
Maybe better not put it in the DTO but outside with `export constant PaymentStatusMapper`?